src/pool/srv_pool.c

```diff
@@ -4619,17 +4638,25 @@ ds_pool_update_handler(crt_rpc_t *rpc)
 	D_DEBUG(DF_DSMS, DF_UUID": processing rpc %p: ntargets=%zu\n",
 		DP_UUID(in->pti_op.pi_uuid), rpc, in->pti_addr_list.ca_count);
 
+	rc = pool_svc_lookup_leader(in->pti_op.pi_uuid, &svc,
+				    &out->pto_op.po_hint);
+	if (rc != 0)
+		goto out;
+
 	list.pta_number = in->pti_addr_list.ca_count;
 	list.pta_addrs = in->pti_addr_list.ca_arrays;
-	rc = ds_pool_update(in->pti_op.pi_uuid, opc_get(rpc->cr_opc), &list,
-			    &inval_list_out, &out->pto_op.po_map_version,
-			    &out->pto_op.po_hint, false);
-	if (rc)
-		D_GOTO(out, rc);
+	rc = pool_svc_update_map(svc, opc_get(rpc->cr_opc),
```

Comment:
When control plane (dmg pool exclude) occurs, this handler I think should be invoked with opcode POOL_EXCLUDE. And if it is excluding a rank (all targets) like pool_svc_exclude_rank() then in this scenario should true be provided as the exlude_rank argument?

Reply:
It seems so. @wangdi1, in which cases do we want to also exclude ONLY all targets but NOT the parent rank? I'm wondering why we don't always do "exclude_rank == true".

Reply:
I think there is such case during hot pull&plug.  @NiuYawei  is that true?

Reply:
@kccain, there are too many different users of this code path. Could we keep the current exclude_rank == false behavior and file a bug for it?

Reply:
Filed https://jira.hpdd.intel.com/browse/DAOS-7273. @wangdi1, could you take that ticket, please? It seems to be a regression from the target exclusion changes.

Reply:
Yes I guess keep exclude_rank = false here in this patch. When I read the patch code I guess I wasn't focused on existing master and see that it always specifies false today. Seems the use of target id = -1 an be one component of determining whether to exclude the rank, though perhaps there is a hot plug case to check also.